tests/test_spider.py

```diff
@@ -92,6 +92,23 @@ def closed(self, reason):
                                        spider=spider, reason=None)
         self.assertTrue(spider.closed_called)
 
+    def test_idled_signal_call(self):
+        class TestSpider(self.spider_class):
+            idled_called = False
+
+            def idled(self):
+                self.idled_called = True
+
+        crawler = get_crawler()
+        spider = TestSpider.from_crawler(crawler, 'example.com')
+        crawler.signals.send_catch_log(signal=signals.spider_opened,
+                                       spider=spider)
+        crawler.signals.send_catch_log(signal=signals.spider_idle,
+                                       spider=spider)
+        crawler.signals.send_catch_log(signal=signals.spider_closed,
+                                       spider=spider, reason=None)
+        self.assertTrue(spider.idled_called)
+
```

Comment:
> So it needs mock.patch.object crawler.engine.crawl (how?) or something else entirely?

I think it is fair to mock `crawler.engine.crawl()` but @kmike prefer to avoid mocks